test/unit/governance/LendingTermOnboarding.t.sol

```diff
@@ -0,0 +1,297 @@
+
+import {Test} from "@forge-std/Test.sol";
+import {Core} from "@src/core/Core.sol";
+import {CoreRoles} from "@src/core/CoreRoles.sol";
+import {MockERC20} from "@test/mock/MockERC20.sol";
+import {GuildToken} from "@src/tokens/GuildToken.sol";
+import {CreditToken} from "@src/tokens/CreditToken.sol";
+import {LendingTerm} from "@src/loan/LendingTerm.sol";
+import {AuctionHouse} from "@src/loan/AuctionHouse.sol";
+import {ProfitManager} from "@src/governance/ProfitManager.sol";
+import {RateLimitedMinter} from "@src/rate-limits/RateLimitedMinter.sol";
+import {LendingTermOnboarding} from "@src/governance/LendingTermOnboarding.sol";
+import {VoltTimelockController} from "@src/governance/VoltTimelockController.sol";
+
+contract LendingTermOnboardingUnitTest is Test {
+    address private governor = address(1);
+    Core private core;
+    ProfitManager private profitManager;
+    GuildToken private guild;
+    CreditToken private credit;
+    MockERC20 private collateral;
+    LendingTerm private termImplementation;
+    AuctionHouse auctionHouse;
+    RateLimitedMinter rlcm;
+    VoltTimelockController private timelock;
+    LendingTermOnboarding private onboarder;
+    address private constant alice = address(0x616c696365);
+    address private constant bob = address(0xB0B);
+
+    // VoltTimelockController params
+    uint256 private constant _TIMELOCK_MIN_DELAY = 3600; // 1h
+
+    // LendingTerm params
+    uint256 private constant _CREDIT_PER_COLLATERAL_TOKEN = 1e18; // 1:1, same decimals
+    uint256 private constant _INTEREST_RATE = 0.05e18; // 5% APR
+    uint256 private constant _HARDCAP = 1_000_000e18;
+
+    // LendingTermOnboarding params
+    uint256 private constant _VOTING_DELAY = 0;
+    uint256 private constant _VOTING_PERIOD = 100_000; // ~14 days
+    uint256 private constant _PROPOSAL_THRESHOLD = 2_500_000e18;
+    uint256 private constant _QUORUM = 20_000_000e18;
+
+    function setUp() public {
+        vm.warp(1679067867);
+        vm.roll(16848497);
+
+        // deploy
+        core = new Core();
+        profitManager = new ProfitManager(address(core));
+        credit = new CreditToken(address(core));
+        guild = new GuildToken(address(core), address(profitManager), address(credit));
+        profitManager.initializeReferences(address(credit), address(guild));
+        collateral = new MockERC20();
+        rlcm = new RateLimitedMinter(
+            address(core), /*_core*/
+            address(credit), /*_token*/
+            CoreRoles.RATE_LIMITED_CREDIT_MINTER, /*_role*/
+            type(uint256).max, /*_maxRateLimitPerSecond*/
+            type(uint128).max, /*_rateLimitPerSecond*/
+            type(uint128).max /*_bufferCap*/
+        );
+        auctionHouse = new AuctionHouse(
+            address(core),
+            650,
+            1800
+        );
+        termImplementation = new LendingTerm();
+        termImplementation.initialize(
```

Comment:
It shouldn't be possible to initialize a logic contract implementation.

Reply:
why ?

The deployed implementation will be initialized already, with all parameters set to 0 (so we know no borrow can happen even if the term were granted the proper roles)

Reply:
it's just good form